elkm1_lib/areas.py

```diff
@@ -126,3 +128,11 @@ def _ld_handler(self, area: int, log: dict[str, Any]) -> None:
             # arm/disarm log (YAGNI - decode number for more log types when needed)
             log["user_number"] = log["number"]
         self.elements[area].setattr("last_log", log, True)
+
+    def _kf_handler(self, keypad: int, key: str, chime_mode: list[int]):
+        for area,mode in enumerate(chime_mode):
+            try:
+                name = ChimeMode(mode).name
+            except ValueError:
+                name = ""
```

Comment:
I don't understand the reasoning for this try/except. Seems to be checking if the `mode` is in range, but then nothing is done with the result of the check.

Reply:
Yes, I wasn't to sure why it was done that way, but I duplicated what is in kc_handler https://github.com/gwww/elkm1/blob/main/elkm1_lib/keypads.py#L56

Reply:
So in keypads kc_handler it uses name:
`        self.elements[keypad].setattr("last_keypress", (name, key), True)`
But in areas kf_handler I didn't return name:
`            self.elements[area].setattr("chime_mode", mode)`
since it only cares about chime_mode in area.

Reply:
So the reason it's different is in kf_decode I handled converting it to an enum, should I change this to just return the int and leave it up to handler to convert to enum?
```
def kf_decode(msg: str) -> dict[str, Any]:
    """KF: Keypad function key press."""
    return {
        "keypad": int(msg[4:6]) - 1,
        "key": FunctionKeys(msg[6]),
        "chime_mode": [ChimeMode(int(x)) for x in msg[7:15]],
    }
```

Reply:
it now passes name thru like your kc_decode.